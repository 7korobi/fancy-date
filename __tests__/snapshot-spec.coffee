{ mount, createLocalVue } = require '@vue/test-utils'
localVue = createLocalVue()
Vuex = require 'vuex'
VueRouter = require 'vue-router'

localVue.use Vuex
localVue.use VueRouter

component = require './index-spec.vue'
router = new VueRouter()
store = new Vuex.Store
  modules:
    index:
      namespaced: true
      state:
        a: "a"
        b: "b"
      mutations:
        update: (state, o)->
          Object.assign state, o
        swap: (state)->
          [state.b, state.a] = [state.a, state.b]

describe "pushState", =>
  test 'snapshot', =>
    wrapper = mount component, { localVue, router, store }
    { vm } = wrapper
    # base state
    expect( vm.$route ).toMatchSnapshot()
    expect( wrapper.html() ).toMatchSnapshot()

    # vuex swap
    store.commit 'index/swap'
    expect( wrapper.html() ).toMatchSnapshot()

    # vuex update
    vm.a = "local_changed_a"
    expect( wrapper.html() ).toMatchSnapshot()

    # state swap
    vm.swap()
    expect( vm.$route ).toMatchSnapshot()
    expect( wrapper.html() ).toMatchSnapshot()
