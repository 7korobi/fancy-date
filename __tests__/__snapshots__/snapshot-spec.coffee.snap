// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pushState snapshot 1`] = `
Object {
  "fullPath": "/",
  "hash": "",
  "matched": Array [],
  "meta": Object {},
  "name": null,
  "params": Object {},
  "path": "/",
  "query": Object {},
}
`;

exports[`pushState snapshot 2`] = `"<div><span>a</span><span>b</span><span>c</span><span>d</span><span>e</span><span>f</span><span>g</span><span>h</span><span>i</span></div>"`;

exports[`pushState snapshot 3`] = `"<div><span>b</span><span>a</span><span>c</span><span>d</span><span>e</span><span>f</span><span>g</span><span>h</span><span>i</span></div>"`;

exports[`pushState snapshot 4`] = `"<div><span>local_changed_a</span><span>a</span><span>c</span><span>d</span><span>e</span><span>f</span><span>g</span><span>h</span><span>i</span></div>"`;

exports[`pushState snapshot 5`] = `
Object {
  "fullPath": "/",
  "hash": "",
  "matched": Array [],
  "meta": Object {},
  "name": null,
  "params": Object {},
  "path": "/",
  "query": Object {
    "c": "f",
    "d": "e",
    "e": "d",
    "f": "c",
  },
}
`;

exports[`pushState snapshot 6`] = `"<div><span>local_changed_a</span><span>a</span><span>f</span><span>e</span><span>d</span><span>c</span><span>i</span><span>h</span><span>g</span></div>"`;
